pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -48,6 +48,16 @@ const (
 	AllValidationsSucceeded = "all validations succeeded"
 )
 
+// BuildReasonPtr returns a pointer to the passed BuildReason.
+func BuildReasonPtr(s BuildReason) *BuildReason {
```

Comment:
Do you think it is possible to omit the `Ptr` suffix? Might be only my preference, but I think it would look better without the suffix. From a usage perspective, it should not make a bit different, one can see the function signature and figure out it being a pointer.

Reply:
I was following the same [pattern](https://github.com/kubernetes/utils/blob/fddb29f9d0095a7a37582a1d86732ae83620f2e8/pointer/pointer.go#L92) as in the `k8s.io/utils/pointer` package. But now I can see that they no longer have the [`Ptr`](https://github.com/kubernetes/utils/blob/587287796c64e9013f0f2374e7e6f546bad92834/pointer/pointer.go#L158) suffix (they may be removed it's because the package name itself have the indication of pointer). If we remove the `Ptr` suffix from our code, we will get a redeclaration error since the package already has the [`BuildReason`](https://github.com/shipwright-io/build/blob/dc719a7b5bfd0f7c66793f08b55de21385a48f0b/pkg/apis/build/v1alpha1/build_types.go#L14) type defined.

Reply:
Well, it is just my preference, but then I would say that this `type BuildReason` should not be exported. Then, we would be able to use `BuildReason` as a name for a function. Also, this would make sure that only we can define a `BuildReason`. At the moment, someone else could make a string into a `BuildReason` since the type is exported.

Reply:
@SaschaSchwarze0 @adambkaplan @otaviof @gabemontero @imjasonh What are your thoughts on this?

Reply:
If we do not export the type, then the constants like `BuildNameInvalid` can still be used I guess? Beside that, what's the best practice in golang? Do argument types of a public function also need to be public ?

Reply:
@SaschaSchwarze0 
>BuildNameInvalid can still be used I guess

Correct

>Do argument types of a public function also need to be public

Not necessary

@HeavyWombat 
`type BuildReason` is imported and used in other packages (See [here]()). So I don't think it's possible to make it private.